### ISSUE TITLE
feat: Usage of hooks in middleware

### ISSUE THREAD

**Issue by mathieutu** [nuxt-community/composition-api#300]
Hi, 
I actually don't know how we could do that, but it would be really cool to:
- be able to use our custom hooks in middleware,
- declare middlewares via the composition api.

What do you think about that?

Would it be something realisable? The First point is the most important for me. For now I've hooks that depends of the store, axios, api calls, etc. but I can't use them in middelwares.

Thanks!

Mathieu.

**Comment by mathieutu** [comment]
Hi @danielroe!
Nothing fancy actually. Just custom hooks that wrap logic from store, axios, route, etc. So the only root hook that we can use here is useContext.

I'll copy/paste you some code example tomorrow if you want.

**Comment by mathieutu** [comment]
Hi! (I'm sorry but I don't work anymore on the project which needed that, so I trying to remember properly the issue)

I think it's what I did actually, but for composed hooks (hooks in hooks), I had to do that in each of them, and so it was not really convenient. And this was for usage in page middleware but same api than here. 

I ended up to replace middlewares by setup function usages. It's not the same, and it's not as good as middleware, but it made the job easily. 

For example : 
```js
export default defineComponent({
  setup() {
    const { redirectToDashboard } = useAuthUtilisateur()

    redirectToDashboard()

    return () => null
  },
})
```
